### PR TITLE
Laura 

### PR DESCRIPTION
**Pull request by lauracodecreations**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, and how did you try querying the API? | I used the Edamam documentation and tested the key and values in Postman
What is an API Wrapper? Why is it in `lib`? How would your project change if you needed to interact with more than one API (aka more than just the Edamam API)? | An API wrapper makes API request and transforms the incoming data into a Ruby hash data. It is in 'lib' folder because it is data for our application. If I need to interact with more than API, I will make a new Wrapper for it.
Describe your API wrapper, the methods you created in it, and why you decided to create those methods/how they were helpful | I have three methods in my API wrapper: list_recipes, find_recipes, and format_uri. I decided to create these methods because they list recipes and find recipes need to be two different API calls. The format uri is useful because it formats the request to search for a recipe in the API 
What was an edge case or failure case test you wrote for your API Wrapper? What was a nominal case? | not yet 
How does VCR aid in testing an API? | not yet
What is the Heroku URL of your deployed application? | have a H10 error https://cheflaura.herokuapp.com/

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Rails.application.routes.draw do
+  root 'recipes#home'
+  get 'index', to: 'recipes#index', as: 'recipes_index'
+  get ':uri/show', to: 'recipes#show', as: 'recipes_show'
```
Comment (dHelmgren) [inline config/routes.rb:4]: Not sure why `:uri` is before show here. Could you use `resources` for the index and show pages?

File: lib/edamam_api_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  def self.format_uri(uri)
+    uri = uri.split("#")
+    base_uri = "http://www.edamam.com/ontologies/edamam.owl#"
+    r = base_uri + uri.last
+    return CGI.escape(r)
+  end
+
+  def self.find_recipe(uri)
```
Comment (dHelmgren) [inline lib/edamam_api_wrapper.rb:34]: nice work finding a way to keep this method DRY

File: test/controllers/recepies_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class RecepiesControllerTest < ActionDispatch::IntegrationTest
+  test "should get index" do
+    get recepies_index_url
+    assert_response :success
+  end
```
Comment (dHelmgren) [inline test/controllers/recepies_controller_test.rb:7]: This test doesn't cover all your cases.

- What happens if no search term is provided?
- What if the search returns no results?

It might also be worthwhile to add some tests around the paging parameters:

- What happens if they're absent?
- Do you get different results when they change?
- What if you send a bogus value, like a negative page number?

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    <script src="https://developer.edamam.com/attribution/badge.js"></script>
+    <title>ApiMuncher</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+  </head>
+
+  <body>
```
Comment (dHelmgren) [inline app/views/layouts/application.html.erb:15]: You should probably have a header that can take the user back to the main page.

File: app/views/recipes/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<section class="channels-list">
+  <% if @recipes != []%>
+    <h1>Recepies for <%=params[:message]%></h1>
+    <ul>
+      <% @recipes.each do |recipe| %>
+      <div class="card" style="width: 18rem;">
+        <img class="card-img-top" src="<%=recipe.image_url%>" alt="Card image cap">
+        <div class="card-body">
+        <h5 class="card-title"<%=recipe.label%></h5>
```
Comment (dHelmgren) [inline app/views/recipes/index.html.erb:9]: Can't see this text on my screen.

**Comment by dHelmgren** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** |
Git hygiene | Good
Comprehension questions | Good
**General** |
Rails fundamentals (RESTful routing, use of named paths) | See comments
Semantic HTML | could be better, see comments
Errors are reported to the user | No, see comments
API Wrapper to handle the API requests | yes
Controller testing | Needs more, see comments
Lib testing | Missing
Search Functionality | yes
List Functionality | yes
Show individual item functionality | yes
**Styling** |
List view shows 10 items at a time and/or has pagination | 10 at a time.
Attractive and usable UI | A little too bear bones. No way to go back to the home page.
**API Features** |
The App attributes Edamam | not on the index page.
The VCR cassettes do not contain the API key | No cassettes.
**External Resources** |
Link to deployed app on Heroku | Broken.
**Overall** | Unfortunately, I don't think you've met the learning goals for this assignment. I would have liked to see your tests, and see you using VCR too before I saw styling. What you have here is a really good start, and I'd like to make sure that you have the support you need to see where you can improve in the future.
